sdk/src/azure/iot/az_iot_hub_client.c

```diff
@@ -61,27 +64,38 @@ AZ_NODISCARD az_result az_iot_hub_client_get_user_name(
   const az_span* const module_id = &(client->_internal.options.module_id);
   const az_span* const user_agent = &(client->_internal.options.user_agent);
   const az_span* const model_id = &(client->_internal.options.model_id);
+  const az_span* const method_twin_ct = &(client->_internal.options.method_twin_content_type);
 
   az_span mqtt_user_name_span
       = az_span_create((uint8_t*)mqtt_user_name, (int32_t)mqtt_user_name_size);
 
   int32_t required_length = az_span_size(client->_internal.iot_hub_hostname)
-      + az_span_size(client->_internal.device_id) + (int32_t)sizeof(hub_client_forward_slash)
+      + (int32_t)sizeof(hub_client_forward_slash) + az_span_size(client->_internal.device_id)
       + az_span_size(hub_service_api_version);
+
   if (az_span_size(*module_id) > 0)
   {
-    required_length += az_span_size(*module_id) + (int32_t)sizeof(hub_client_forward_slash);
+    required_length += (int32_t)sizeof(hub_client_forward_slash) + az_span_size(*module_id);
   }
   if (az_span_size(*user_agent) > 0)
   {
-    required_length += az_span_size(*user_agent) + az_span_size(hub_client_param_separator_span);
+    required_length += az_span_size(hub_client_param_separator_span) + az_span_size(*user_agent);
```

Comment:
Why swap the order of operations here. This change seems unnecessary (unless I am missing some logical difference).

Reply:
It's not necessary technically, but the swapped order now matched the order the username will be built in.  It was harder to envision what was being measured when it was in the backwards order.

Reply:
Gotcha, that makes sense. I just wanted to make sure it wasn't some compiler optimization/re-ordering issue.